homeassistant/components/aussie_broadband/__init__.py

```diff
@@ -27,7 +28,7 @@ async def async_setup_entry(
     # Login to the Aussie Broadband API and retrieve the current service list
     client = AussieBB(
         entry.data[CONF_USERNAME],
-        entry.data[CONF_PASSWORD],
+        SecretStr(entry.data[CONF_PASSWORD]),
```

Comment:
Ideally our code does not depend on pydantic. HA does not rely on pydantic iirc, so we should not reference them here

Reply:
Does that mean I need to fork the library and shim the init?

We can't roll back the library change because that was done in the Pydantic migration.

Reply:
Why can't we change the library?

Reply:
It's not maintained by either code owner, we can raise a PR and see what happens.

Reply:
Alternatively, could we use duck typing here?

Would just need a wrapper around the string ie:

```python


class SecretStr:
  """Fake secret string to avoid dependency on Pydantic."""
  def __init__(self, value: string):
    this._value = value
  
  def get_secret_value(self):
    return self._value
```

Reply:
https://github.com/yaleman/pyaussiebb/pull/233 raised, if published I'll change this PR to a bump.